qa/tasks/cephfs/filesystem.py

```diff
@@ -383,9 +387,11 @@ def __init__(self, ctx, fscid=None, create=None):
                 self.name = create
             if not self.legacy_configured():
                 self.create()
-        elif fscid is not None:
-            self.id = fscid
-        self.getinfo(refresh = True)
+                self.getinfo(refresh = True)
+        else:
+            if fscid is not None:
+                self.id = fscid
+            self.name = name
```

Comment:
`fscid` and `name` should be mutually exclusive options. If `fscid` is provided, we need to do `self.getinfo(refresh=True)`.

I think I'd be happier if the fs was left unnamed and just use a flag like `defer_create` so the meaning is clear. Then use `self.set_name` to set the name prior to create as you do in `cephfs_test_case.py`.